ducktape/cluster/remoteaccount.py

```diff
@@ -247,8 +247,12 @@ def ssh(self, cmd, allow_fail=False):
         stdout.read()
         exit_status = stdout.channel.recv_exit_status()
         try:
-            if not allow_fail and exit_status != 0:
-                raise RemoteCommandError(self, cmd, exit_status, stderr.read())
+            if exit_status != 0:
+                if not allow_fail:
+                    raise RemoteCommandError(self, cmd, exit_status, stderr.read())
+                else:
+                    msg = "Running ssh command exited with status %d and message '%s': %s" % (exit_status, stderr.read(), cmd)
```

Comment:
I would rearrange it to print the command line prior to the stderr output since the latter can be multiline.

Reply:
Sorry, can you explain?

Reply:
oh, I think I understand.

Reply:
If the command fails and prints multiple lines of output to stderr the debug log will be like:
```
Running ssh command exited with status 2 and message 'Failed to bla bla
because here is an exception:
  function1
  method2
  otherthing
and some/other/stuff/maybe': /path/to/a/program/that/fails --some args
```

and I personally find that harder to read than prefixing the output with the command:
```
Running ssh command '/path/to/a/program/that/fails --some args' exited with status 2: Failed to bla bla
because here is an exception:
  function1
  method2
  otherthing
and some/other/stuff/maybe'
```


But this is a tiny tiny nit pick of personal preference, it may be ignored :)